controllers/metering/controllers/metering_controller.go

```diff
@@ -275,10 +279,151 @@ func NewBillingList(TimeInterval meteringv1.TimeIntervalType, amount int64) mete
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	err = r.Delete(ctx, &metering)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	return nil
+}
+
+func (r *MeteringReconcile) GetAllExtensionResources(ctx context.Context) (map[corev1.ResourceName]meteringv1.ResourcePriceAndUsed, error) {
+	extensionResourcesPriceList := meteringv1.ExtensionResourcePriceList{}
+	totalResourcePrice := make(map[corev1.ResourceName]meteringv1.ResourcePriceAndUsed, 0)
+	err := r.List(ctx, &extensionResourcesPriceList)
+	if err != nil {
+		return totalResourcePrice, client.IgnoreNotFound(err)
+	}
+	for _, extensionResourcesPrice := range extensionResourcesPriceList.Items {
+		for k, v := range extensionResourcesPrice.Spec.Resources {
+			totalResourcePrice[k] = meteringv1.ResourcePriceAndUsed{ResourcePrice: v, Used: &resource.Quantity{}}
+		}
+	}
+	return totalResourcePrice, nil
+}
+
+func (r *MeteringReconcile) syncAccountBalance(ctx context.Context, owner string, amount int64, seqID int64) error {
+	if amount == 0 {
+		return nil
+	} else if amount < 0 {
+		return fmt.Errorf("deduction amount is <0")
+	}
+
+	accountBalance := userv1.AccountBalance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s-%v", userv1.AccountBalancePrefix, owner, seqID),
+			Namespace: r.MeteringSystemNameSpace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, &accountBalance, func() error {
+		accountBalance.Spec.Owner = owner
+		accountBalance.Spec.TimeStamp = time.Now().Unix()
+		accountBalance.Spec.Amount = amount
+		accountBalance.Status.Status = meteringv1.Create
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *MeteringReconcile) SetupWithManager(mgr ctrl.Manager) error {
+	const controllerName = "metering-controller"
+	r.Logger = ctrl.Log.WithName(controllerName)
+	r.Logger.V(1).Info("reconcile controller metering")
+
+	// get env METERING_SYSTEM_NAMESPACE and METERING_INTERVAL
+	r.MeteringSystemNameSpace = os.Getenv(METERINGNAMESPACEENV)
+	if os.Getenv(METERINGNAMESPACEENV) == "" {
+		r.MeteringSystemNameSpace = DEFAULTMETERINGNAMESPACE
+	}
+
+	timeInterval := os.Getenv("METERING_INTERVAL")
+
+	if timeInterval == "" || timeInterval == "0" {
+		timeInterval = DefaultInterval()
+	}
+	r.MeteringInterval, _ = strconv.Atoi(timeInterval)
+	if r.MeteringInterval == 0 {
+		r.MeteringInterval = DefaultIntervalInt()
+	}
+	r.Logger.Info("metering env", METERINGNAMESPACEENV, r.MeteringSystemNameSpace, "timeinterval:", r.MeteringInterval, "envtimeinterval", os.Getenv("METERING_INTERVAL"))
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&meteringv1.Metering{}).
 		Watches(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{}).
+		Watches(&source.Kind{Type: &meteringv1.ExtensionResourcePrice{}}, &handler.EnqueueRequestForObject{}).
+		Watches(&source.Kind{Type: &meteringv1.Resource{}}, &handler.EnqueueRequestForObject{}).
```

Comment:
IMO, why not split these watchs to defirent controllers?

Reply:
What does IMO mean

Reply:
because one CR should changed by one controller，if split controller will lead to 2 controller change same CR，lead to same confilct。